stdlib/lang-parser/parserModule/parser.bal

```diff
@@ -0,0 +1,1236 @@
+// Copyright (c) 2019 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/io;
+import ballerina/log;
+
+type Parser object {
+    //stack which holds the operators during expression rule
+    OperatorStack operatorStack = new;
+    //fill the operator map
+    OperatorMapper operatorMapper = new;
+    //fill value type map
+    ValueTypeMapper valueTypeMapper = new;
+    //stack which holds the expression for the expression rule
+    ExpressionStack expStack = new;
+    //keeps track on error recovery
+    boolean errorRecovered = true;
+    //keep track of error tokens
+    Token[] errorTokens = [];
+    //error token count
+    int errorCount = 0;
+    //keeps track on next expected token should be an operator or operand
+    boolean expectOperand = true;
+    //if invalid occurrence is captured in an expression
+    boolean invalidOccurence = false;
+    // tuple list expression count
+    int tupleListPos = 0;
+    //counter in the operator token in list in tuple literal expression
+    int separatorCount = 0;
+    //there is a operator in the prior
+    boolean priorOperator = false;
+	//buffer which store the tokens
+    private ParserBufferReader parserBuffer;
+
+    public function __init(ParserBufferReader parserBuffer) {
+        self.parserBuffer = parserBuffer;
+    }
+
+    public function parse() returns PackageNode {
+        DefinitionNode?[] defList = [];
+        //definition list position
+        int pos = 0;
+        //parse while the current token is not EOF token
+        while (!self.parserBuffer.isEOFToken()) {
+            Token currToken = self.parserBuffer.consumeToken();
+
+            if (currToken.tokenType == FUNCTION) {
+                FunctionNode functionNode = self.parseFunction(currToken);
+                DefinitionNode defNode = functionNode;
+                defList[pos] = defNode;
```

Comment:
Do we need L62? 

Why can't we directly say 
```suggestion
                defList[pos] = functionNode;
```

Reply:
Ballerina lang-parser implementation in no longer maintained and has already being removed from the ballerina-lang repo.